rust/connlib/snownet/src/node.rs

```diff
@@ -565,13 +565,18 @@ where
     ) -> Connection {
         agent.handle_timeout(self.last_now);
 
+        /// We set a Wireguard keep-alive to ensure the WG session doesn't timeout on an idle connection.
+        ///
+        /// Without such a timeout, using a tunnel after the REKEY_TIMEOUT requires handshaking a new session which delays the new application packet by 1 RTT.
+        const WG_KEEP_ALIVE: Option<u16> = Some(10);
```

Comment:
Is this WireGuard's persistent_keepalive setting? Won't setting this prevent connections from ever going stale? Or is snownet/str0m smart enough to differentiate between WG keepalives and application data transmission in order to kill idle connections?

Just thinking back to the discussion on #3807

Reply:
We don't yet have functionality for killing idle connections but once we implement that, yes we will be able to differentiate between application data and keep-alives.

I would have implemented keep-alive tracking by checking when we last either read packet from the tunnel or wrote one into it. Basically, record when we last _attempted_ to use the tunnel (attempt because writing to the tunnel doesn't guarantee that it is delivered). As long as we set the idle timeout longer than the ICE timeout, this should work because we will run into the ICE timeout before and kill the connection as not working.